application-licensing-licensor/application-licensing-licensor-api/src/main/java/com/xwiki/licensing/LicensingConfiguration.java

```diff
@@ -72,4 +72,17 @@
      * @return the email of the licensing owner or null if the value of the property is not filled up
      */
     String getLicensingOwnerEmail();
+
+    /**
+     * @return the list of extensions that have a new version available, that users have been notified of
+     */
+    List<String> getNewVersionNotifiedExtensions();
+
+    /**
+     * Set the list of extensions with a new version available, that users have been notified of.
+     *
+     * @param newVersionNotifiedExtensions the list of extensions with a new version available, that users have been
+     *            notified of
+     */
+    void setNewVersionNotifiedExtensions(List<String> newVersionNotifiedExtensions);
```

Comment:
I don't see how this is related to configuration, which normally includes settings that the administrators can set to tweak the behavior of the application. In this case, I don't see why the administrators would set themselves the lists of extensions that have a new version available. This looks more like some **internal** state that the licensing application needs to manage in order to prevent duplicate notifications. The administrators shouldn't be aware of this.

Reply:
What would be the best way of saving this internal state? Maybe have a separate page that holds an object with the list of extensions?

Reply:
First of all, keeping all the versions for all the extensions doesn't scale (if we don't do any cleanup). The list grows indefinitely. I think we could keep only the last **known** installable version for each extension and namespace (that we have notified the administrators of). Each time this last know version changes we send a new notification. So I'm thinking we could use a class with 3 fields (namespace, extension id and version) and add objects of this type to the licensing config page, making sure we have only one object per namespace and extension id. This should be "hidden" behind a component that gives you the last known installable version and allows you to update it, in case we want to implement it differently in the future.